test/e2e/framework/globalegressip.go

```diff
@@ -0,0 +1,85 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package framework
+
+import (
+	"context"
+	"fmt"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+)
+
+var globalEgressIPGVR = &schema.GroupVersionResource{
+	Group:    "submariner.io",
+	Version:  "v1",
+	Resource: "globalegressips",
+}
+
+func globalEgressIPClient(cluster ClusterIndex, namespace string) dynamic.ResourceInterface {
+	return DynClients[cluster].Resource(*globalEgressIPGVR).Namespace(namespace)
+}
+
+func CreateGlobalEgressIP(cluster ClusterIndex, obj *unstructured.Unstructured) error {
+	geipClient := globalEgressIPClient(cluster, obj.GetNamespace())
+	AwaitUntil("create GlobalEgressIP", func() (interface{}, error) {
+		egressIP, err := geipClient.Create(context.TODO(), obj, metav1.CreateOptions{})
+		if apierrors.IsAlreadyExists(err) {
+			err = geipClient.Delete(context.TODO(), obj.GetName(), metav1.DeleteOptions{})
+			if err != nil {
+				fmt.Printf("Returning error while deleting: %v", err)
+				return nil, err
+			}
+
+			egressIP, err = geipClient.Create(context.TODO(), obj, metav1.CreateOptions{})
+		}
```

Comment:
Do we really need "CreateAnew" functionality?

Reply:
Ideally, we don't because each test creates a new namespace, but if a test-case is written in a manner where it restarts certain pods and tries to modify the object it will achieve what it desires. Also, since we are not comparing if the existing object matches with the one we are creating, personally I feel there is no harm in the current logic. As you know, if the object does not exist, it will be as good as simply calling create.

Reply:
It just seems to me we should let the test control the lifecycle instead of making assumptions here.

Reply:
Okay, I'll update it. Thanks.